src/discounts/components/DiscountDates/DiscountDatesWithController.tsx

```diff
@@ -0,0 +1,29 @@
+import { DiscoutFormData } from "@dashboard/discounts/types";
+import React, { ChangeEvent } from "react";
+import { useController } from "react-hook-form";
+
+import DiscountDates from "./DiscountDates";
+
+export const DiscountDatesWithController = () => {
+  const { field } = useController<DiscoutFormData, "dates">({
+    name: "dates",
+  });
+
+  const handleChange = (e: ChangeEvent<any>) => {
+    field.onChange({
+      ...field.value,
+      [e.target.name]:
+        e.target.name === "hasEndDate" ? e.target.checked : e.target.value,
+    });
```

Comment:
suggestion: actually that can be in separated function. Its name can indicate what exactly you generate (readability)